joomla/joomla-cms
# [4.2] Fix code logic issues in Categories model

Pull Request for Issue # .

### Summary of Changes
In frontend Categories model of our components (com_content in this PR), there are few logic issues:

- `$params` in `getItems` method use params from the active menu item instead of calculated params stored in model states. With code like that, it will have wrong value for parameters set to **Use Global** in the menu item parameters (as it Use Global, it should get value from component Options but that's not working here because we are getting params directly from active menu item at the moment).

-  The returned categories list does not depend on value of two model states `filter.access` and `filter.published`. So if someone uses this model class and want to get list of unpublished categories (by setting `filter.published` state to 0, it won't work)

This PR fixes the two above logic issues in the code. If it is accepted, the similar changes should be done for other categories.

### Testing Instructions
This is just code logic issues and does not have much effect how the system works, so there is not anything difference which testers can see. The only thing you can test is apply PR, create a menu item to link to **List All Categories in an Article Category Tree** and see that it is still working as expected.

I hope someone could review code, too.
